sdk/tests/client/addresses.rs

```diff
@@ -259,5 +278,39 @@ async fn search_address() -> Result<()> {
 
     assert_eq!(res, (9, false));
 
-    Ok(())
+    // Internal
+    let addresses = secret_manager
+        .generate_ed25519_addresses(
+            GetAddressesOptions::from_client(&client)
+                .await?
+                .internal()
+                .with_coin_type(IOTA_COIN_TYPE)
+                .with_account_index(0)
+                .with_range(9..10)
+                .with_bech32_hrp(IOTA_BECH32_HRP),
+        )
+        .await?;
+
+    let res = iota_sdk::client::api::search_address(
+        &secret_manager,
+        IOTA_BECH32_HRP,
+        IOTA_COIN_TYPE,
+        0,
+        0..10,
+        &addresses[0],
+    )
+    .await?;
+
+    assert_eq!(res, (9, true));
+
+    // not in range
+    let res =
+        iota_sdk::client::api::search_address(&secret_manager, IOTA_BECH32_HRP, IOTA_COIN_TYPE, 0, 0..9, &addresses[0])
+            .await;
+
+    if let Err(iota_sdk::client::Error::InputAddressNotFound { .. }) = res {
+        return Ok(());
+    }
+
+    panic!("should not have found search address range & public")
```

Comment:
This is a bit weird tbh, can you just assert that res is the err please?

Reply:
Cant,. Error isnt PartialEq. ill put it in an else statement to make it slightly better

Reply:
Don't think you need PartialEq if you use `matches!`